raft/fsm.hh

```diff
@@ -27,7 +27,7 @@ struct fsm_output {
     std::vector<std::pair<server_id, rpc_message>> messages;
     // Entries to apply.
     std::vector<log_entry_ptr> committed;
-    std::optional<applied_snapshot> snp;
+    std::vector<applied_snapshot> snps;
```

Comment:
we couldn't just skip taking a snapshot in `applier_fiber` since it'll complicate the [backpressure logic](https://github.com/scylladb/scylladb/pull/11481/commits/01728c2541e211c8c6b5cd67498daf564b3339c6#diff-7c8eecebd17f6e9f8791f2e135952aa7e6b39430d95816a30325926c3692b4c8R540) even further

Reply:
You can't just point to a code and expect me to understand why it was even written. If there is a problem explain it in words.

Reply:
Also nobody proposed skipping taking a snapshot. Or at least not that I remember it.